### PR TITLE
Doc nits for bpo-16500

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

**Review by serhiy-storchaka — COMMENTED** [review]
`PyOS_AfterFork()` also is mentioned in `Doc/c-api/init.rst`.

**Comment by pitrou** [comment]
Thanks @serhiy-storchaka, for some reason I had overlooked it.

File: Doc/c-api/init.rst [hunk 1]
```diff
@@ -564,7 +564,7 @@ Additionally, when extending or embedding Python, calling :c:func:`fork`
 directly rather than through :func:`os.fork` (and returning to or calling
 into Python) may result in a deadlock by one of Python's internal locks
 being held by a thread that is defunct after the fork.
-:c:func:`PyOS_AfterFork` tries to reset the necessary locks, but is not
+:c:func:`PyOS_AfterFork_Child` tries to reset the necessary locks, but is not
```
Comment (serhiy-storchaka) [inline Doc/c-api/init.rst:567]: Is it worth to update the first part of this paragraph according to new API?

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

But maybe `Doc/c-api/init.rst` needs more updates.

File: Doc/c-api/init.rst (hunk 1 above)
Comment (pitrou) [reply Doc/c-api/init.rst:567]: Which part exactly?

Comment (serhiy-storchaka) [reply Doc/c-api/init.rst:567]: "there is no way to inform Python of additional (non-Python) locks that need to be acquired before or reset after a fork" and mentioning `pthread_atfork`.

**Pull request merged**